From 92755ccd1f263a54d816ccbd7b32c65c481b6c47 Mon Sep 17 00:00:00 2001
From: K <kayo@illumium.org>
Date: Tue, 23 Aug 2022 12:09:47 +0500
Subject: [PATCH 2/2] Quick fix for bl31.elf has a LOAD segment with RWX
 permissions

After upgrading binutils on debian testing (2.38.50.20220707-1 to 2.38.90.20220713-2) atf started to fail building with the following warning, causing an error:

    $ make BUILD_BASE=build CROSS_COMPILE=aarch64-linux-gnu- bl31
    ...
      LD      /home/martinet/code/trusted-firmware-a/build/fvp/release/bl31/bl31.elf
    aarch64-linux-gnu-ld.bfd: warning: /home/martinet/code/trusted-firmware-a/build/fvp/release/bl31/bl31.elf has a LOAD segment with RWX permissions
    make: *** [Makefile:1128: /home/martinet/code/trusted-firmware-a/build/fvp/release/bl31/bl31.elf] Error 1
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3941f8698..4b82781a8 100644
--- a/Makefile
+++ b/Makefile
@@ -435,7 +435,7 @@ TF_LDFLAGS		+=	$(subst --,-Xlinker --,$(TF_LDFLAGS_$(ARCH)))
 
 # LD = gcc-ld (ld) or llvm-ld (ld.lld) or other
 else
-TF_LDFLAGS		+=	--fatal-warnings -O1
+TF_LDFLAGS		+=	--fatal-warnings -O1 --no-warn-rwx-segments
 TF_LDFLAGS		+=	--gc-sections
 # ld.lld doesn't recognize the errata flags,
 # therefore don't add those in that case
-- 
2.36.0

